grpc/protogen.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+# Points to the directory of the file whose //go:generate
+# directive is calling this script.
+PROTO_DIR=$(pwd)
+# Lists all .proto files in the directory.
+PROTO_FILES=$(ls $PROTO_DIR/*.proto)
```

Comment:
```suggestion
declare -a PROTO_FILES=()
for file in ${PWD}/*
do
    [ -e "${file}" ] || continue
    PROTO_FILES+=("${file}")
done
```

While parsing `ls` is common, it's a bad practice with a lot of pitfalls. A better approach is to create an empty array and rely on file globing and `test` to append files to that array.

The following will still produce the nicely space separated list of `.proto` files with a full path.
```shell
echo "${PROTO_FILES[@]}"
```

Reply:
Switched to a loop over the results of `find`, PTAL.